config/config_test.go

```diff
@@ -26,6 +26,126 @@ import (
 	v3_3 "github.com/coreos/ignition/v2/config/v3_3_experimental/types"
 )
 
+// helper to check whether a type and field matches a denylist of known problems
+// examples are either structs or names of structs
+func ignore(t reflect.Type, field reflect.StructField, fieldName string, examples ...interface{}) bool {
+	if field.Name != fieldName {
+		return false
+	}
+	for _, candidate := range examples {
+		if reflect.TypeOf(candidate).Kind() == reflect.String {
+			if t.Name() == candidate.(string) {
+				return true
+			}
+		} else if t == reflect.TypeOf(candidate) {
+			return true
+		}
+	}
+	return false
+}
+
+// vary the specified field value and check the given key function to see
+// whether the field seems to affect it
+// this function's heuristic can be fooled by complex key functions but it
+// should be fine for typical cases
+func fieldAffectsKey(key func() string, v reflect.Value) bool {
+	kind := v.Kind()
+	switch {
+	case util.IsPrimitive(kind):
+		old := key()
+		v.Set(util.NonZeroValue(v.Type()))
+		new := key()
+		v.Set(reflect.Zero(v.Type()))
+		return old != new
+	case kind == reflect.Ptr:
+		null := key()
+		v.Set(reflect.New(v.Type().Elem()))
+		allocated := key()
+		affectsKey := fieldAffectsKey(key, v.Elem())
```

Comment:
Just to make sure I'm reading the golang docs right, this line is specifically testing if changing the underlying interface (because we would have already tested `nil` & empty struct) affects the key?

Reply:
This is testing whether pointing to a non-zero value affects the key, yeah.